services/121-service/src/payments/fsp-integration/commercial-bank-ethiopia/commercial-bank-ethiopia.service.ts

```diff
@@ -68,16 +65,15 @@ export class CommercialBankEthiopiaService
     private readonly transactionsService: TransactionsService,
     @Inject(REDIS_CLIENT)
     private readonly redisClient: Redis,
+    public readonly programFspConfigurationRepository: ProgramFinancialServiceProviderConfigurationRepository,
   ) {}
 
   public async sendPayment(
     paPaymentList: PaPaymentDataDto[],
     programId: number,
     paymentNr: number,
   ): Promise<FspTransactionResultDto> {
-    const credentials: { username: string; password: string } =
-      await this.getCommercialBankEthiopiaCredentials(programId);
-
+    // ##TODO: Regactor this to ensure that the programFinancialServiceProviderConfigurationId is the same for the whole list
```

Comment:
The idea is that at the end of this refactor we can do ctr + shift + f to find all of the comments relevant for this refactor (and not the others). Than we can decide if we forgot something. And maybe also remove the ## of comment that we decide not to do straight away.

Reply:
That sounds pretty convenient. So we do not want to merge any "##TODO" to main. Is this a common practice in our team? If not, maybe it could? And add an automated check for it for merging to main, if possible?